CONTRIBUTING.md

```diff
@@ -1,10 +1,27 @@
-# Submitting Issues
+# Contributing Guide
 
-* Issues can be submitted in two different places:
+## Submitting Issues
+
+* Issues can be submitted in two different places. <b>Please search for your issues in both places before submitting new issues:</b>
     1) [The GitHub issue log](https://github.com/shadowsocks/shadowsocks-android/issues)
     2) [The Shadowsocks.org Discourse Forum](https://discourse.shadowsocks.org/c/implementations/shadowsocks-android)
-* Please search for your issues in both places before submitting new issues.
 * If you have any questions, please check the [FAQ](https://github.com/shadowsocks/shadowsocks-android/blob/master/.github/faq.md) and the questions submitted at the [Discourse Forum](https://discourse.shadowsocks.org/c/implementations/shadowsocks-android).
 * General discussions should go to the [Discourse Forum](https://discourse.shadowsocks.org/c/implementations/shadowsocks-android).
-* Always provide information asked in the template unless you know what you're doing.
-* Utilizing Logcat would also be helpful.
+* Always provide the information requested in the related [template](https://github.com/shadowsocks/shadowsocks-android/tree/master/.github/ISSUE_TEMPLATE) when submitting issues.  Please note that the templates are provided when submitting on GitHub, but you will need to copy the correct template from the link provided above when submitting to the [Discourse Forum](https://discourse.shadowsocks.org/c/implementations/shadowsocks-android).
+* For reporting bugs, utlizing [Logcat](https://developer.android.com/studio/command-line/logcat) and providing the related error text would also be helpful.
+
+## Making Contributions
+
+* Make sure that the contribution you plan to make has been logged as an issue.
+* Fork the repo, clone it and code!  NOTE:
+    * Please limit fixes to the issue at hand.  Make separate pull requests for other issues.
+    * Comment your code as needed for clarity.
+    * Utilize descriptive variable names, single purpose functions and other clean coding practices.
+    * Use descriptive commit and PR comments.
+    * Test and build locally before submitting a pull request.
+* When ready, create a pull request that clearly indicates the issue this is relating to and any other pertinent information.
+* Wait for review and respond to any follow-up.  
+* Thank you for contributing!
```

Comment:
This seems like a very generic guide, which I guess is also fine.

Reply:
I was hoping to put more detail, but I could not find some of what I was looking for.  For example, I can see mention of a linter in a few places throughout the code, but I could not decipher what set up was being used.  If I missed it somewhere, I'd be happy to add more detail.

Reply:
Lint is used by Codacy. You can find a link in README. The lint should be followed as much as possible but sometimes it is permissible to have a violation due to detekt bug or other reasons.

Reply:
Added lint information to contributing guide.